src/main/kotlin/io/kni/thingoo/backend/readings/NewReadingMqttMessageHandler.kt

```diff
@@ -0,0 +1,30 @@
+package io.kni.thingoo.backend.readings
+
+import io.kni.thingoo.backend.mqtt.MqttMessageHandler
+import io.kni.thingoo.backend.readings.dto.SaveReadingDto
+import org.springframework.stereotype.Component
+
+@Component
+class NewReadingMqttMessageHandler(
+    private val readingService: ReadingService
+) : MqttMessageHandler {
+
+    companion object {
+        val entityNewReadingTopicRegex = Regex("^/devices/(\\w+)/entities/(\\w+)/reading$")
+    }
+
+    override fun handle(message: String, topic: String) {
+        val (deviceKey, entityKey) = getDestructuredTopic(topic)
+
+        val reading = SaveReadingDto(
+            value = message,
+            entityKey = entityKey,
+            deviceKey = deviceKey
+        )
+        readingService.saveReading(reading)
+    }
+
+    private fun getDestructuredTopic(topic: String): MatchResult.Destructured {
```

Comment:
name feels a little awkward when looking at the usage above, not really sure but perhaps something like `getKeysFromTopic` would be a bit better?

Reply:
Resolved in https://github.com/ThingooKNI/backend/pull/59/commits/0a11ba51836846ae170ce1633a141f0f05e8dc57